source/adapters/opencl/kernel.cpp

```diff
@@ -189,11 +190,45 @@ urKernelGetSubGroupInfo(ur_kernel_handle_t hKernel, ur_device_handle_t hDevice,
     InputValueSize = MaxDims * sizeof(size_t);
   }
 
-  cl_int Ret = clGetKernelSubGroupInfo(cl_adapter::cast<cl_kernel>(hKernel),
-                                       cl_adapter::cast<cl_device_id>(hDevice),
-                                       mapURKernelSubGroupInfoToCL(propName),
-                                       InputValueSize, InputValue.get(),
-                                       sizeof(size_t), &RetVal, pPropSizeRet);
+  // We need to allow for the possibility that this device runs an older CL and
+  // supports the original khr subgroup extension.
+  using ApiFuncT =
+      cl_int(CL_API_CALL *)(cl_kernel, cl_device_id, cl_kernel_sub_group_info,
+                            size_t, const void *, size_t, void *, size_t *);
+  ApiFuncT GetKernelSubGroupInfo = nullptr;
+
+  oclv::OpenCLVersion DevVer;
+  CL_RETURN_ON_FAILURE(cl_adapter::getDeviceVersion(
+      cl_adapter::cast<cl_device_id>(hDevice), DevVer));
+
+  if (DevVer < oclv::V2_1) {
+    bool SubgroupExtSupported = false;
+
+    UR_RETURN_ON_FAILURE(cl_adapter::checkDeviceExtensions(
+        cl_adapter::cast<cl_device_id>(hDevice), {"cl_khr_subgroups"},
+        SubgroupExtSupported));
+    if (!SubgroupExtSupported) {
+      return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
+    }
+    cl_platform_id Platform;
+    CL_RETURN_ON_FAILURE(clGetDeviceInfo(
+        cl_adapter::cast<cl_device_id>(hDevice), CL_DEVICE_PLATFORM,
+        sizeof(Platform), &Platform, nullptr));
+    GetKernelSubGroupInfo =
+        reinterpret_cast<ApiFuncT>(clGetExtensionFunctionAddressForPlatform(
```

Comment:
We should use the extension function pointer cache for this.

Reply:
Huh yeah forgot we had those, I cribbed from `ProgramCreateWithIL` when I wrote this since it's a similar cl version situation there, updated that as well